src/lib/polkadotApi.ts

```diff
@@ -1,30 +1,35 @@
 import { ApiPromise, Keyring, WsProvider } from '@polkadot/api';
 import { getBlockNumber, getEpoch } from './connections';
-import { dotApi, storeBlockNumber, storeConnected, storeToken } from './stores';
+import { dotApi } from './stores';
 import { options } from '@frequency-chain/api-augment';
+import { storeChainInfo } from './stores';
 
-import type { DotApi, MsaInfo } from '$lib/storeTypes';
+import type { DotApi, MsaInfo } from '../lib/storeTypes';
 import type { KeyringPair } from '@polkadot/keyring/types';
 import type { ChainProperties } from '@polkadot/types/interfaces';
 import type { Option, u64 } from '@polkadot/types';
 
 export type AccountMap = Record<string, KeyringPair>;
 
-export async function getApi(selectedProviderURI: string, thisDotApi: DotApi, wsProvider: WsProvider) {
-  if (!selectedProviderURI) {
-    throw new Error('Empty providerURI');
-  }
-  // Handle disconnects
-  if (selectedProviderURI) {
-    if (thisDotApi?.api) {
-      await thisDotApi.api.disconnect();
-    } else if (wsProvider) {
-      await wsProvider.disconnect();
-    }
-  }
+async function updateChainInfo(api: ApiPromise) {
+  const chain = await api.rpc.system.properties();
+  const token = getToken(chain);
+  const blockNumber = (await getBlockNumber(api)) as bigint;
+  const epochNumber = await getEpoch(api);
+  storeChainInfo.set({ connected: true, blockNumber, epochNumber, token });
+}
+
+export async function createAndConnectToApi(networkEndpoint: string) {
+  dotApi.subscribe(async (apiState: DotApi) => await apiState.api?.disconnect());
```

Comment:
I don't understand this line. When the dotApi store changes we trigger a disconnect?

Reply:
@wilwade The way I understood it, we want to disconnect the old API instance before creating a new instance. Shannon's code worked the same way.

Reply:
Ended up deleting. It was causing problems.